integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/datamap/CarbonCreateDataMapCommand.scala

```diff
@@ -128,7 +140,17 @@ case class CarbonCreateDataMapCommand(
       if (mainTable != null && !deferredRebuild) {
         dataMapProvider.rebuild()
         if (dataMapSchema.isIndexDataMap) {
+          val operationContext: OperationContext = new OperationContext()
+          val systemFolderLocation: String = CarbonProperties.getInstance().getSystemFolderLocation
+          val createDataMapPreExecutionEvent: CreateDataMapPreExecutionEvent =
+            new CreateDataMapPreExecutionEvent(sparkSession, systemFolderLocation)
+          OperationListenerBus.getInstance().fireEvent(createDataMapPreExecutionEvent,
+            operationContext)
           DataMapStatusManager.enableDataMap(dataMapName)
+          val createDataMapPostExecutionEvent: CreateDataMapPostExecutionEvent =
+            new CreateDataMapPostExecutionEvent(sparkSession, systemFolderLocation)
+          OperationListenerBus.getInstance().fireEvent(createDataMapPostExecutionEvent,
+            operationContext)
```

Comment:
here it should be UpdateDataMapPreExecutionEvent and UpdateDataMapPostExecutionEvent instead of createDataMapPreExecutionEvent and createDataMapPostExecutionEvent